### PR TITLE
chore: move next admin packages to core repo

### PR DESCRIPTION
**Pull request by kasperkristensen**
**What**
- Move packages for `next` version of admin to core repo

**Other**
- Since this PR introduces packages that depend on Vite 5, it also introduces @types/node@^20. We have never had a direct dependency on the types package for Node, and as far as I can see that has resulted in us using the types from Node.js@8, as those are a dependency of one of our dependencies. With the introduction of @types/node@^20, two of our packages had TS errors because they were using the NodeJS.Timer type, which was deprecated in Node.js@14. We should add specific @types/node packages to all our packages, but I haven't done so in this PR to keep it as clean as possible.
- Q: @olivermrbl I've added the new packages to the ignore list for changeset, is this enough to prevent them from being published?

### CODE REVIEW

**Comment by olivermrbl** [comment]
> I've added the new packages to the ignore list for changeset, is this enough to prevent them from being published?

Yes, albeit with two caveats as per the changesets docs:
```
There are two caveats to this.

- If the package is mentioned in a changeset that also includes a package that is not ignored, publishing will fail.
- If the package requires one of its dependencies to be updated as part of a publish.
```

But I think we can manage.

Alternatively, we do as with the modules packages and just never include them in changesets.

Or thirdly, add `private: true` to the `package.json` 😄

File: packages/admin-next/admin-bundler/src/api/build.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { resolve } from "path";
```
Comment (olivermrbl) [inline packages/admin-next/admin-bundler/src/api/build.ts]: suggestion: remove semi-colon

File: packages/admin-next/admin-bundler/src/api/create-vite-config.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+      transform: { ...content1.transform, ...content2.transform },
+    }
+  } else {
+    throw new Error("Cannot merge content fields of different types")
+  }
+
+  const mergedConfig = deepmerge(config1, config2)
+  mergedConfig.content = mergedContent
+
+  console.log(config1.presets, config2.presets)
```
Comment (olivermrbl) [inline packages/admin-next/admin-bundler/src/api/create-vite-config.ts:85]:
```suggestion
```

File: packages/admin-next/admin-bundler/src/api/dev.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { createServer } from "vite";
```
Comment (olivermrbl) [inline packages/admin-next/admin-bundler/src/api/dev.ts]: suggestion: remove semi-colon

File: packages/admin-next/dashboard/public/locales/en/translation.json [hunk 4]
```diff
@@ -0,0 +1,106 @@
+{
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/public/locales/en/translation.json:1]: praise: nice, can we make do with this few translations? if so, love it!

File: packages/admin-next/dashboard/src/app.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import { AuthProvider } from "./providers/auth-provider";
+import { RouterProvider } from "./providers/router-provider";
+import { ThemeProvider } from "./providers/theme-provider";
+
+import { queryClient } from "./lib/medusa";
+
+function App() {
+  return (
+    <MedusaProvider
+      baseUrl="http://localhost:9000"
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/app.tsx]: question: Should this be configurable?

File: packages/admin-next/dashboard/src/components/authentication/require-auth/require-auth.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  if (auth.isLoading) {
+    return (
+      <div className="min-h-screen flex items-center justify-center">
+        <Spinner className="animate-spin text-ui-fg-interactive" />
+      </div>
+    );
+  }
+
+  if (!auth.user) {
+    console.log("redirecting");
```
Comment (olivermrbl) [inline packages/admin-next/dashboard/src/components/authentication/require-auth/require-auth.tsx:20]:
```suggestion
```

**Review by olivermrbl — APPROVED** [review]

**Comment by olivermrbl** [comment]
I think you might be missing prettier files to eliminate the semi-colons

**Comment by olivermrbl** [comment]
Is this a long-running PR? I thought the idea was to merge this, so we can do the follow-up changes in "reviewable" chunks. Happy to do either approach.

File: packages/admin-next/dashboard/public/locales/en/translation.json (hunk 4 above)
Comment (kasperkristensen) [reply packages/admin-next/dashboard/public/locales/en/translation.json:1]: Planning on doing most if not all the translations we currently have + adding danish. Before release I'll run this file through chat gpt, for each translation and ping the team for reviews of the ones we can check ourselves.

**Comment by kasperkristensen** [comment]
> Is this a long-running PR? I thought the idea was to merge this, so we can do the follow-up changes in "reviewable" chunks. Happy to do either approach.

Merging now, I just didn't want to stop work until this was merged, but moving forward each piece of work will be one PR.

**Pull request merged**